### DISCUSSION TITLE
Meili map matching skips links

### DISCUSSION THREAD

**Discussion by JohanKlaas** [valhalla/valhalla#3224] [Q&A]
* I'm new to Valhalla and map matching, so please bear with me. 

I'm trying to use a local version of Valhalla-Meili to map-match a bunch of hand-drawn routes to the OSM network. I prepare my data using Python and submit it to Valhalla using POST / trace_route as recommended in this tutorial (mostly just adjusting costing to "pedestrian"): https://towardsdatascience.com/map-matching-done-right-using-valhallas-meili-f635ebd17053

However, the results which I obtain in places deviate strongly from from OSM network. This is most apparent in the screen print below. 

- Indicated in RED are the lines which represent my input data. 
- Indicated in BLUE are the lines generated by Valhalla-Meili. 

_Clearly, the dark blue lines indicate routes which are utterly impossible:_ 
- On the left we see Valhalla projecting a route through a building (rather than following the path which the original data is quite close to. 
- On the right we see Valhalla projecting a line across a waterway where there is no bridge.

**Do you see any way by which I can force Valhalla to match my routes onto the actual OSM network?** 

_Attempted solutions:_
(1) I have tried toggling the "search_radius" and "max_search_radius" fields, but these do not seem to have any effect.
(2) I have tried increasing the coordinate density of the input data (i.e. splitting up each edge into several evenly sized segments), in case the issue was caused by outliers, but this had little effect either. 
(3) I tried switching the costing mode to "bicycle". This delivered a slightly better result, although still far from perfect. 

![Screenshot 2021-07-21 at 16 32 39](https://user-images.githubusercontent.com/87767866/126506517-30c5e4f3-3332-4857-8798-b3e13c0d384b.png)

**Comment by kevinkreiser** [comment]
are you sure you are decoding the shape returned in the map match correctly? maybe you can paste the verbatim response you got back from the service or at least the part of it in the `shape` field?
